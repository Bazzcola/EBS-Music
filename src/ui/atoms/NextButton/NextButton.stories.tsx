import React from 'react';
import { NextButton } from 'ui/atoms/NextButton/NextButton';

export default {
  title: 'Atoms/NextButton',
  component: NextButton
};
export const Button3 = () => {
  return <NextButton />;
};
