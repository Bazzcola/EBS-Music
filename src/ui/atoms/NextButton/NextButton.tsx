import React, { useState } from 'react';
import 'ui/atoms/NextButton/NextButton.scss';

export const NextButton = () => {
  const [active, setActive] = useState<boolean>(false);
  const changes = () => {
    setActive(!active);
  };
  return (
    <button onClick={changes} className="next_button">
      <svg className="svg_nextbtn" viewBox="0 0 12 12">
        <path d="M11 .5v11c0 .276-.232.5-.5.5a.503.503 0 0 1-.5-.5V7.08L1.562 11.95A.375.375 0 0 1 1 11.624V.376c0-.29.312-.47.562-.325L10 4.919V.5c0-.276.232-.5.5-.5.276 0 .5.229.5.5z"></path>
      </svg>
    </button>
  );
};
