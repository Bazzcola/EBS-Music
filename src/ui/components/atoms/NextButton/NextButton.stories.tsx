import React from 'react';
import { NextButton } from 'ui/components/atoms/NextButton/NextButton';

export default {
  title: 'Atoms/NextButton',
  component: NextButton
};
export const Button3 = () => {
  return <NextButton />;
};
