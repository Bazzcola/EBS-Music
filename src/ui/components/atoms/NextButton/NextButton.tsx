import React, { useState, useContext, useEffect, useCallback } from 'react';
import { AudioContext, song } from 'ui/context/audioContext';
import { Tracks } from 'ui/context/audioContext';
import './NextButton.scss';

export const NextButton = () => {
  const [files, setFiles] = useState<string[]>(['']);
  const {
    counter,
    setCounter,
    setPlaying,
    setAudioFiles,
    profile
  } = useContext(AudioContext);

  useEffect(() => {
    setFiles(Tracks.map((item: song) => item.src));
  }, []);

  let keys: any = Object.keys(files);

  const nextSong = useCallback(
    (e: { preventDefault: () => void }) => {
      e.preventDefault();
      if (profile === false) {
        if (counter === files.length - 1) {
          setCounter(0);
          setAudioFiles(files[keys[0]]);
        } else {
          setCounter((count: number) => count + 1);
          setAudioFiles(files[keys[counter + 1]]);
          setPlaying(true);
        }
      } else {
        return;
      }
    },
    [counter, files, keys, profile, setAudioFiles, setCounter, setPlaying]
  );
  return (
    <button
      onClick={nextSong}
      className={`next_button ${profile ? 'next_button_active' : ''}`}
    >
      <svg className="svg_nextbtn" viewBox="0 0 12 12">
        <path d="M11 .5v11c0 .276-.232.5-.5.5a.503.503 0 0 1-.5-.5V7.08L1.562 11.95A.375.375 0 0 1 1 11.624V.376c0-.29.312-.47.562-.325L10 4.919V.5c0-.276.232-.5.5-.5.276 0 .5.229.5.5z"></path>
      </svg>
    </button>
  );
};
