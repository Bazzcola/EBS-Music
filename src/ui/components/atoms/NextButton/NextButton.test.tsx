import React from 'react';
import { render } from '@testing-library/react';
import { NextButton } from './NextButton';

it('render NextButton correct', () => {
  const button = () => {
    render(<NextButton />);
  };
  expect(button).toMatchSnapshot();
});
