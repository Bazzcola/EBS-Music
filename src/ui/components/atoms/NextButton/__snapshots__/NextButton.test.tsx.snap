// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NextButton correct 1`] = `[Function]`;
